opacus/grad_sample/grad_sample_module.py

```diff
@@ -341,15 +351,19 @@ def capture_backprops_hook(
         grad_sampler_fn = self.GRAD_SAMPLERS[type(module)]
         grad_samples = grad_sampler_fn(module, activations, backprops)
         for param, gs in grad_samples.items():
-            create_or_accumulate_grad_sample(param=param, grad_sample=gs, layer=module)
+            create_or_accumulate_grad_sample(
+                param=param, grad_sample=gs, max_batch_len=module.max_batch_len
+            )
+
+        for p in module.parameters():
+            p._forward_counter -= 1
+            if p._forward_counter == 0:
+                promote_current_grad_sample(p)
```

Comment:
nit: I suppose this can still be inside the `if len(module.activations) == 0` block?

Reply:
The check itself - yes, but the decrement - no.